core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1132,6 +1158,38 @@ public VersionNumber getNeededDependenciesRequiredCore() {
             return versionNumber;
         }
 
+        /**
+         * Returns the minimum Java version needed to use the plugin and all its dependencies.
+         *
+         * @return the minimum Java version needed to use the plugin and all its dependencies.
+         *
+         * @since TODO
+         */
+        @CheckForNull
+        public VersionNumber getNeededDependenciesMinimumJavaVersion() {
```

Comment:
The name of the method is tricky for me. Do you like instead, something like **getMinimumJavaVersionConsideringDependencies** or similar? :pray:

Reply:
I tend to agree, but I used the existing naming pattern. So I think either we'd fix everything, or nothing.

Reply:
Nothing to be consistent then.

Reply:
In *this* PR, I agree adjusting everything would be out of scope. But fixing it in a dedicated PR could make sense.